moabb/evaluations/evaluations.py

```diff
@@ -35,6 +43,36 @@
 Vector = Union[list, tuple, np.ndarray]
 
 
+def objective(trial, X, y, clf, scorer, epochs):
+    learning_rate = trial.suggest_float("learning_rate", 1e-4, 1e-2, log=True)
+    weight_decay = trial.suggest_float("weight_decay", 1e-10, 1e-3, log=True)
+    drop_rate = trial.suggest_float("drop_rate", 0.3, 0.9)
+
+    pre_model, model = create_deep_model(
+        clf, learning_rate, weight_decay, drop_rate, epochs=epochs
+    )
+    n_epochs = list(range(len(y)))
+    try:
+        idx_X_train, idx_X_val, y_train, y_val = train_test_split(
```

Comment:
With this, you won't have the same validation set at each iteration, making it hard to compare the different HP.
You should fix a seed.

Reply:
fixed, can you take a look again @tomMoral?